inductiva/tasks/task.py

```diff
@@ -169,18 +170,92 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
 
-            if status in _TASK_TERMINAL_STATUSES:
+            if self._is_terminal_status:
                 return status
 
             time.sleep(polling_period)
 
-    def kill(self) -> None:
-        """Kill the task.
+    def _is_terminal_status(self) -> bool:
+        """Check if the task is in a terminal status.
+
+        This method issues a request to the API.
+        """
+        status = self.get_status()
+        return status in _TASK_TERMINAL_STATUSES
+
+    def kill(
+            self,
+            wait_timeout: Optional[Union[float,
+                                         int]] = None) -> Union[bool, None]:
+        """Request a task to be killed.
+        
+        This method requests that the current task is remotely killed.
+        If `wait_timeout` is None (default), the kill request is sent to the
+        backend and the method returns. However, if `wait_timeout` is 
+        a positive number, the method waits up to `wait_timeout` seconds
+        to ensure that the task transitions to the KILLED state.
+        Args:
+            wait_timeout (int, float): Optional - number of seconds to wait for the
+            kill command or None if only the request is to be sent.
+        Returns:
+            - None if `wait_timeout` is None and the kill request was
+              successfully sent;
+            - True if `wait_timeout`> 0 and the task successfully transitioned
+              to the `KILLED` state within `wait_timeout` seconds;
+            - False if `wait_timeout` > 0 but the task didn't transition
+              to the `KILLED` state within `wait_timeout` seconds;
 
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
         """
-        self._api.kill_task(path_params=self._get_path_params())
+        if wait_timeout is not None:
+            if not isinstance(wait_timeout, (float, int)):
+                raise TypeError("Wait timeout must be a number.")
+            if wait_timeout <= 0.0:
+                raise ValueError("Wait timeout must be a positive number.")
+
+        maxretries = constants.TASK_KILL_MAX_API_REQUESTS
+
+        while maxretries > 0:
+            maxretries -= 1
+            try:
+                if self._is_terminal_status():
+                    break
+
+                path_params = self._get_path_params()
+                self._api.kill_task(path_params=path_params)
+                break
+            except exceptions.ApiException as exc:
+                if maxretries == 0:
+                    raise RuntimeError(
+                        "Something went wrong while sending"
+                        " the kill command. Please try again later.") from exc
+                time.sleep(constants.TASK_KILL_RETRY_SLEEP_SEC)
+
+        if wait_timeout is None:
+            return None
+
+        success = True
+        start = time.time()
+        status = self.get_status()
```

Comment:
no need for the assignment. The walrus in the `while` condition already performs the assignment.